### ISSUE TITLE
[RFE] Build/use rclone

### ISSUE THREAD

**Issue by ScottESanDiego** [fsantini/KoboCloud#34]
This works great, thanks for putting this together!

Unfortunately we have `curl` getting the file every time (we should look at using `-z` in `curl` to compare against the old file time), and limited support for cloud sources.

Updating to use `rclone` would allow more diverse support of sources, as well as authenticated support.  E.g., have a user create an rclone config file, then copy it over to kobo.

(Assumes a working golang to build EABI5 binaries, of course - I do not have such a thing.)

Reference: [https://rclone.org/](https://rclone.org/)

**Comment by ScottESanDiego** [comment]
See PR35.

**Comment by fsantini** [comment]
Hi Scott!
Great improvement, indeed rclone seems the way to go, it would solve a lot of maintenance problems.
I suggest we don't try to implement rclone alongside the current curl, but we make a new branch and eventually replace the whole package with the rclone version.

**Comment by ScottESanDiego** [comment]
Yeah, I too think that just ripping it all out and using rclone is the real right answer, but I didn't want to get that radical in the PR and break everyone already using kobocloud.  It certainly would simplify things though!
